src/DTPPodNode.cpp

```diff
@@ -67,14 +66,27 @@ namespace dunedaq {
     const OutputSinkNode& DTPPodNode::get_output_sink_node() const {
       return getNode<OutputSinkNode>("outsink");
     }
-    
 
-    void DTPPodNode::reset() const {
-    
+    std::map<std::string, uint32_t> DTPPodNode::get_firmware_config_info() const {
+      auto info_node = get_info_node();
+      auto l_config_names = info_node.getNode("config").getNodes();
+      std::map<std::string, uint32_t> l_firmware_info;
+      for (auto& l_name : l_config_names) {
+	std::cout << l_name <<std::endl;
+	auto temp = info_node.getNode("config."+l_name).read();
+	getClient().dispatch();
+	l_firmware_info.insert({l_name, temp});
+	}
+      return l_firmware_info;
+    }
+
+    void DTPPodNode::reset() {
+      m_dtp_fw_info = get_firmware_config_info();
+      m_n_links = m_dtp_fw_info["n_links"];
```

Comment:
i think this feature (retrieving number of links from FW, rather than config) should be discussed before we add it.  typically we would only use this info to check/confirm consistency of the dunedaq config with the firmware.

also, the value of m_n_links will currently depend on whether reset() has been called, or not.  that's guaranteed to cause problems.  i suggest we set to zero in ctor, then the true value in configure, and not in reset().  there can be a method which checks consistency between the 'software' configuration and firmware one.